packages/client/src/settings/types/getSettings.ts

```diff
@@ -0,0 +1,27 @@
+import type { Config } from '../../index.js';
+
+type PasswordRules = {
+  rules: [
+    {
+      name: string;
+      regex: string;
+    },
+  ];
+  differFromLastPass: number;
+};
+
+export type SettingsQuery = {
+  channelCode?: string;
+  type: string;
+};
+
+export type Settings = {
+  channelCode?: string;
+  type: string;
+  details: PasswordRules | object;
```

Comment:
Should this be a generic type?
Having `object` there is not very useful to consumers. This `details` property can be anything the user wants or is it restricted? If it can be anything the user wants, then I would change this to a generic (and change all clients/actions/selectors that depend on it).

Reply:
For now only the Password Rules is useful for usage in the client, there are static models for that values but we don't want the clients to use them, so we just declare as an object to avoid typescript error.